src/mango/src/mango_cursor_view.erl

```diff
@@ -383,22 +381,23 @@ view_cb({row, Row}, #mrargs{extra = Options} = Acc) ->
             % However, this oddness is confined to being visible in this module.
             case match_and_extract_doc(Doc, Selector, Fields) of
                 {match, FinalDoc} ->
-                    FinalViewRow = ViewRow#view_row{doc = FinalDoc},
-                    ok = rexi:stream2(FinalViewRow),
+                    ViewRow = fabric_view_row:set_doc(ViewRow0, FinalDoc),
```

Comment:
Suggestion: might be simpler  to use `ViewRow1` here and keep `ViewRow` everywhere else, instead of having `ViewRow0` everywhere else.

The typical pattern is to use `Var0` for the first few lines in a function,  prior to transforming it into `Var` near the top of the function and then use Var. This is usually for validation and such. Then `Var` would be the most used value and then if it's updated it's `Var1`, `Var2`, ...